cdap-api/src/main/java/co/cask/cdap/api/TransactionUtil.java

```diff
@@ -0,0 +1,215 @@
+    try {
+      transactional.execute(new TxRunnable() {
+        @Override
+        public void run(DatasetContext context) throws Exception {
+          runnable.run(context);
+        }
+      });
+    } catch (TransactionFailureException e) {
+      throw propagate(e, exception);
+    }
+  }
+
+  /**
+   * Executes the given {@link TxCallable} using the given {@link Transactional}.
+   *
+   * @param transactional the {@link Transactional} to use for transactional execution.
+   * @param callable the {@link TxCallable} to be executed inside a transaction
+   * @param <V> type of the result
+   * @return value returned by the given {@link TxCallable}.
+   * @throws  RuntimeException if failed to execute the given {@link TxRunnable} in a transaction.
+   * If the TransactionFailureException has a cause in it, the cause is propagated.
+   */
+  public static <V> V execute(Transactional transactional,
+                              final TxCallable<V> callable) {
+    final AtomicReference<V> result = new AtomicReference<>();
+    try {
+      transactional.execute(new TxRunnable() {
+        @Override
+        public void run(DatasetContext context) throws Exception {
+          result.set(callable.call(context));
+        }
+      });
+    } catch (TransactionFailureException e) {
+      throw propagate(e);
+    }
+    return result.get();
+  }
+
+  /**
+   * Executes the given {@link TxCallable} using the given {@link Transactional}.
+   *
+   * @param transactional the {@link Transactional} to use for transactional execution.
+   * @param callable the {@link TxCallable} to be executed inside a transaction
+   * @param <V> type of the result
+   * @return value returned by the given {@link TxCallable}.
+   * @throws X if failed to execute the given {@link TxRunnable} in a transaction. If the TransactionFailureException
+   * has a cause in it, the cause is thrown as-is if it is an instance of X.
+   */
+  public static <V, X extends Throwable> V execute(Transactional transactional,
+                              final TxCallable<V> callable, Class<X> exception) throws X {
+    final AtomicReference<V> result = new AtomicReference<>();
+    try {
+      transactional.execute(new TxRunnable() {
+        @Override
+        public void run(DatasetContext context) throws Exception {
+          result.set(callable.call(context));
+        }
+      });
+    } catch (TransactionFailureException e) {
+      throw propagate(e, exception);
+    }
+    return result.get();
+  }
+
+  /**
+   * Propagates {@code throwable} as-is if it is an instance of
+   * {@link RuntimeException} or {@link Error}, or else as a last resort, wraps
+   * it in a {@code RuntimeException} then propagates.
+   * @param throwable the Throwable to propagate
+   * @return nothing will ever be returned; this return type is only for convenience
+   */
+  public static RuntimeException propagate(Throwable throwable) {
```

Comment:
@chtyim Its called from multiple places, but only in this class for now. This is because I am adding the execute() methods to call both execute and propagate. 
But propagate() could be called directly from outside the class. Just like the existing Transactions class [which is also public]